draft-ietf-taps-arch.md

```diff
@@ -126,32 +126,37 @@ they appear in all capitals, as shown here.
 
 - Applications create connections and transfer data using the socket API.
-- The socket API provides the interface to the implementations of TCP and UDP (typically implemented in the system's kernel).
-- TCP and UDP in the kernel send and receive data over the available network layer interfaces.
+- The socket API provides the interface to the implementations of TCP and UDP
+  (typically implemented in the system's kernel).
+- TCP and UDP in the kernel send and receive data over the available network
+  layer interfaces.
+- Sockets are bound directly to transport-layer and network-layer addresses,
+  obtained via a separate resolution step, usually performed by a
+  system-provided stub resolver.
 
 ~~~~~~~~~~
 
 +-----------------------------------------------------+
 |                    Application                      |
 +-----------------------------------------------------+
-           |                             |
-+---------------------+       +-----------------------+
-|  Socket Stream API  |       |  Socket Datagram API  |
-+---------------------+       +-----------------------+
-           |                             |
-+-----------------------------------------------------+
-|         TCP                           UDP           |
-|           Kernel Protocol Implementation            |
-+-----------------------------------------------------+
-                          |
+        |                 |                  |
+  +------------+     +------------+    +--------------+
+  |    stub    |     | Stream API |    | Datagram API |
+  |  resolver  |     +------------+    +--------------+
+  +------------+          |                  |
+        |           +---------------------------------+
+        |           |     TCP                UDP      |
+        |           |  Kernel Protocol Implementation |
+        |           +---------------------------------+
+        |                           |
 +-----------------------------------------------------+
 |               Network Layer Interface               |
 +-----------------------------------------------------+
 
 ~~~~~~~~~~
 {: #fig-sockets title="socket() API Model"}
 
-The Transport Services architecture maintains this general model of interaction, but aims to both modernize the API surface exposed for transport protocols and enrich the capabilities of the transport system implementation.
+The Transport Services architecture maintains this general model of interaction, but aims to both modernize the API surface exposed for transport protocols and enrich the capabilities of the transport system implementation. It combines interfaces for multiple interaction patterns into a unified whole. By combining identifier address resoulution with connection establishment and data transfer in a single API, it allows for more flexible implementations to provide path and transport protocol agility on the application's behalf.
```

Comment:
Reading this again now and seeing the figures, I wonder if 

> The Transport Services architecture maintains this general model of interaction, but aims to both modernize the API surface 

may lead the reader to thinking that the Transport System Implementation runs in the kernel? Separate issue I guess, but a bit realted to the bypass comment above.

Reply:
I tweaked this language slightly, but would point out that nothing in the *architecture* precludes a kernel TAPS implementation.